lib/munge.js

```diff
@@ -9,6 +9,53 @@ const mungeNpm = require("./mungers/npm")
 const mungeSh = require("./mungers/sh")
 const mungeShebang = require("./mungers/shebang")
 
+/**
+ * @typedef {object} InternalSpawnOptions Options for the internal spawn functions
+ *   `childProcess.ChildProcess.prototype.spawn` and `process.binding('spawn_sync').spawn`.
+ *   These are the options mapped by the `munge` function to intercept spawned processes and
+ *   handle the wrapping logic.
+ *
+ * @property {string} file File to execute: either an absolute system-dependent path or a
+ *   command name.
+ * @property {string[]} args Command line arguments passed to the spawn process, including argv0.
+ * @property {string | undefined} cwd Optional path to the current working directory passed to the
+ *   spawned process. Default: `process.cwd()`
+ * @property {boolean} windowsHide Boolean controlling if the process should be spawned as
+ *   hidden (no GUI) on Windows.
+ * @property {boolean} windowsVerbatimArguments Boolean controlling if Node should preprocess
+ *   the CLI arguments on Windows.
+ * @property {boolean} detached Boolean controlling if the child process should keep its parent
+ *   alive or not.
+ * @property {string[]} envPairs Array of serialized environment variable key/value pairs. The
+ *   variables serialized as `key + "=" + value`.
+ * @property {import("child_process").StdioOptions} stdio Stdio options, with the same semantics
+ *   as the `stdio` parameter from the public API.
+ * @property {number | undefined} uid User id for the spawn process, same as the `uid` parameter
+ *   from the public API.
+ * @property {number | undefined} gid Group id for the spawn process, same as the `gid` parameter
+ *   from the public API.
+ *
+ * @property {string | undefined} basename Custom property only used by `spawn-wrap`. It is the
+ *   basename of the file to spawn (so individual mungers don't have to duplicate the code to
+ *   compute it).
+ */
+
+/**
+ * Updates the internal spawn options to redirect the process through the shim and wrapper.
+ *
+ * This works on the options passed to `childProcess.ChildProcess.prototype.spawn` and
+ * `process.binding('spawn_sync').spawn`.
+ *
+ * This function works by trying to identify the spawn process and map the options accordingly.
+ * `spawn-wrap` recognizes most shells, Windows `cmd.exe`, Node and npm invocations; when spawn
+ * either directly or through a script with a shebang line.
+ * It also unconditionally updates the environment variables so bare `node` commands execute
+ * the shim script instead of Node's binary.
+ *
+ * @param workingDir {string} Absolute system-dependent path to the directory containing the shim files.
+ * @param options {InternalSpawnOptions} Internal spawn options to update.
+ * @return {void} This function does not return any value, the options are modified in-place.
```

Comment:
Does JSDocs require that we provide `@return` when nothing is returned?

Reply:
No, this is not required but I prefer to add it and explain which values are modified when there is no result.